stdlib/public/SDK/Foundation/Data.swift

```diff
@@ -24,54 +24,773 @@ internal func __NSDataInvokeDeallocatorFree(_ mem: UnsafeMutableRawPointer, _ le
+            }
+        }
+    }
+    
+    public var length: Int {
+        @inline(__always)
+        get {
+            switch _backing {
+            case .swift:
+                return _length
+            case .immutable:
+                return _length
+            case .mutable:
+                return _length
+            case .customReference(let d):
+                return d.length
+            case .customMutableReference(let d):
+                return d.length
+            }
+        }
+        @inline(__always)
+        set {
+            setLength(newValue)
+        }
+    }
+    
+    
+    public func _freeBytes() {
+        if let bytes = _bytes {
+            free(bytes)
+        }
+    }
+    
+    
+    public func enumerateBytes(_ block: (_ buffer: UnsafeBufferPointer<UInt8>, _ byteIndex: Data.Index, _ stop: inout Bool) -> Void) {
+        var stop: Bool = false
+        switch _backing {
+        case .swift:
+            block(UnsafeBufferPointer<UInt8>(start: _bytes?.assumingMemoryBound(to: UInt8.self), count: _length), 0, &stop)
+            break
+        case .immutable:
+            block(UnsafeBufferPointer<UInt8>(start: _bytes?.assumingMemoryBound(to: UInt8.self), count: _length), 0, &stop)
+            break
+        case .mutable:
+            block(UnsafeBufferPointer<UInt8>(start: _bytes?.assumingMemoryBound(to: UInt8.self), count: _length), 0, &stop)
+            break
+        case .customReference(let d):
+            d.enumerateBytes { (ptr, range, stop) in
+                var stopv = false
+                let bytePtr = ptr.bindMemory(to: UInt8.self, capacity: range.length)
+                block(UnsafeBufferPointer(start: bytePtr, count: range.length), range.length, &stopv)
+                if stopv {
+                    stop.pointee = true
+                }
+            }
+            break
+        case .customMutableReference(let d):
+            d.enumerateBytes { (ptr, range, stop) in
+                var stopv = false
+                let bytePtr = ptr.bindMemory(to: UInt8.self, capacity: range.length)
+                block(UnsafeBufferPointer(start: bytePtr, count: range.length), range.length, &stopv)
+                if stopv {
+                    stop.pointee = true
+                }
+            }
+            break
+        }
+    }
     
-    var __wrapped : _MutableUnmanagedWrapper<ImmutableType, MutableType>
+    @inline(never)
+    public func _grow(_ newLength: Int, _ clear: Bool) {
+        let cap = _capacity
+        var additionalCapacity = (newLength >> (_DataStorage.vmOpsThreshold <= newLength ? 2 : 1))
+        if Int.max - additionalCapacity < newLength {
+            additionalCapacity = 0
+        }
+        var newCapacity = Swift.max(cap, newLength + additionalCapacity)
+        let origLength = _length
+        var allocateCleared = clear && _DataStorage.shouldAllocateCleared(newCapacity)
+        var newBytes: UnsafeMutableRawPointer? = nil
+        if _bytes == nil {
```

Comment:
Suggestion: I recommend using `guard let` (and `if let`) a lot more in this function.